axelor-crm/src/main/java/com/axelor/apps/crm/service/OpportunityServiceImpl.java

```diff
@@ -67,6 +69,8 @@ public Partner createClientFromLead(Opportunity opportunity) throws AxelorExcept
           addressService.createAddress(
               null, null, lead.getPrimaryAddress(), null, addressL6, lead.getPrimaryCountry());
       address.setFullName(addressService.computeFullName(address));
+
+      email = new EmailAddress(lead.getEmailAddress().getAddress());
```

Comment:
The if condition is checking primaryAddress, but not email address, currently this could throw a NPE.